software/mk/Makefile.builddefs

```diff
@@ -186,13 +187,17 @@ RISCV_GXX_OPTS += $(RISCV_GXX_EXTRA_OPTS)
 spmd_defs = -DPREALLOCATE=0 -DHOST_DEBUG=0
 
 ifdef CLANG
-LLVM_DIR         ?= $(BSG_MANYCORE_DIR)/software/riscv-tools/llvm/llvm-install
-LLVM_CLANG       ?= $(LLVM_DIR)/bin/clang
-LLVM_OPT         ?= $(LLVM_DIR)/bin/opt
-LLVM_LLC         ?= $(LLVM_DIR)/bin/llc
-PASS_DIR         ?= $(BSG_MANYCORE_DIR)/software/manycore-llvm-pass
-PASS_LIB         ?= build/manycore/libManycorePass.so
-RUNTIME_FNS      ?= $(BSG_MANYCORE_DIR)/software/bsg_manycore_lib/bsg_tilegroup.h
+LLVM_DIR          ?= $(BSG_MANYCORE_DIR)/software/riscv-tools/llvm/llvm-install
+LLVM_CLANG        ?= $(LLVM_DIR)/bin/clang
+LLVM_CLANGPP      ?= $(LLVM_DIR)/bin/clang++
+LLVM_OPT          ?= $(LLVM_DIR)/bin/opt
+LLVM_LLC          ?= $(LLVM_DIR)/bin/llc
+CLANG_TARGET_OPTS ?= --target=riscv32 -march=$(ARCH_OP) -mabi=$(ABI)
+ifdef LLVM_PASS
+PASS_DIR          ?= $(BSG_MANYCORE_DIR)/software/manycore-llvm-pass
```

Comment:
Is LLVM_PASS documented anywhere?

Related: Will we support it in the future?

Reply:
Not sure about this pass, but I imagine we could be adding passes. 

Yes, without documentation, this is more of easter egg than a feature. What do you think is nice place to describe these?

Reply:
In a README. Perhaps in the software directory. 

If we're not planning to use the old LLVM pass in the future, I think we should remove it. However, it does serve as a useful example of an LLVM Pass, until you and Emily add more. 

My suggestion is rename it to something less generic than LLVM_PASS, LLVM_SHARED_MEM_PASS, and document it.

As a general thought though, if we plan to have multiple passes I think that Makefile variables aren't a good way to specify which should/shouldn't be run. It's difficult to specify what order the passes are run in. Is there another way we can specify the passes we want to run, and what order? Ideally, this is per-program

Reply:
Readme in software directory sounds good!

I agree with you regarding passes. The option I added feels more like enable/disable all passes. With our current understanding though, we don't think we could customize scheduler with out-of-source passes. So, let's defer this per-pass enable/disable problem?

Reply:
Yes, file an issue, write it in your list of someday items, or ...

Oh, and change the flag name. Make it reflect what it actually does

Reply:
Great! Yes, I'm about to change the flag name to `ENABLE_LLVM_PASSES`!

Reply:
Added [readme](https://github.com/bespoke-silicon-group/bsg_manycore/blob/clang_cpp/software/README.md).